frontend/pages/_document.jsx

```diff
@@ -16,6 +16,15 @@ class MyDocument extends Document {
             type="image/svg"
             href="../../static/favicon/favicon.ico"
           ></link>
+          <script
+            async
+            src="https://www.googletagmanager.com/gtag/js?id=UA-175933629-1"
+          ></script>
+          <script>
+            window.dataLayer = window.dataLayer || []; function gtag()
+            {dataLayer.push(arguments)}
+            gtag('js', new Date()); gtag('config', 'UA-175933629-1');
+          </script>
```

Comment:
It shows internal server error. I beleive we need to use this snippet 
``` <script
    dangerouslySetInnerHTML={{
      __html: `
            window.dataLayer = window.dataLayer || [];
            function gtag(){dataLayer.push(arguments);}
            gtag('js', new Date());
            gtag('config', '[Tracking ID]');
        `,
    }}
  />
```
[Link](https://stackoverflow.com/questions/60411351/how-to-use-google-analytics-with-next-js-app) to the question on StackOverflow
 
more infot about `dangerouslySetInnerHTML` [link](https://reactjs.org/docs/dom-elements.html#dangerouslysetinnerhtml)